diff --git a/libmamba/src/core/link.cpp b/libmamba/src/core/link.cpp
index 46e344e..a2587bd 100644
--- a/libmamba/src/core/link.cpp
+++ b/libmamba/src/core/link.cpp
@@ -716,4 +716,4 @@ namespace mamba
-        if (ec)
+        if (ec || status != 0)
         {
             LOG_ERROR << "response code: " << status << " error message: " << ec.message();
             if (script_file != nullptr && util::get_env("CONDA_TEST_SAVE_TEMPS"))
